run_test.sh

```diff
@@ -380,6 +381,50 @@ EOF
 sig_key${MODULE_SERIAL}
 EOF
     fi
+
+    # If the key can't be extracted from module, the hash algorithm is also unknown
+    if [[ "${SIG_KEY// /}" != "sig_key:" ]] && grep -q '^CONFIG_MODULE_SIG_HASH=' "/lib/modules/${KERNEL_VER}/build/.config"; then
+        echo 'Building the test module using a different hash algorithm'
+        if grep -q '^CONFIG_MODULE_SIG_HASH="sha512"' "/lib/modules/${KERNEL_VER}/build/.config"; then
+            ALTER_HASH="sha256"
+        else
+            ALTER_HASH="sha512"
+        fi
```

Comment:
Seems a bit odd to see "Building..." where the first thing immediately after is uninstalling.
Maybe move the above echo + if block, further down - just above the 'echo "CONFIG_..." > /tmp/...' line?

Reply:
The installing and uninstalling were removed. They were originally to workaround a bug. I'll open a new issue for that bug.